test/jdk/tools/jpackage/macosx/HostArchPkgTest.java

```diff
@@ -0,0 +1,95 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathFactory;
+import jdk.jpackage.test.JPackageCommand;
+import jdk.jpackage.test.PackageTest;
+import jdk.jpackage.test.PackageType;
+import jdk.jpackage.test.TKit;
+import jdk.jpackage.test.Annotations.Test;
+
+/**
+ * Test will generation default pkg package and will validate "hostArchitectures"
+ * attribute in unpacked pkg package inside Distribution file. See JDK-8266179.
+ */
+
+/*
+ * @test
+ * @summary jpackage test to validate "hostArchitectures" attribute
+ * @library ../helpers
+ * @build jdk.jpackage.test.*
+ * @modules jdk.jpackage/jdk.jpackage.internal
+ * @compile HostArchPkgTest.java
+ * @requires (os.family == "mac")
+ * @key jpackagePlatformPackage
+ * @run main/othervm/timeout=360 -Xmx512m jdk.jpackage.test.Main
+ *  --jpt-run=HostArchPkgTest
+ */
+public class HostArchPkgTest {
+
+    private static void verifyHostArch(JPackageCommand cmd) throws Exception {
+        Path distributionFile = cmd.pathToUnpackedPackageFile(Path.of("/"))
+                .toAbsolutePath()
+                .getParent()
+                .resolve("data")
+                .resolve("Distribution");
+
+        DocumentBuilderFactory dbf
+                = DocumentBuilderFactory.newDefaultInstance();
+        dbf.setFeature("http://apache.org/xml/features/" +
+                       "nonvalidating/load-external-dtd", false);
+        DocumentBuilder b = dbf.newDocumentBuilder();
+        org.w3c.dom.Document doc
+                = b.parse(Files.newInputStream(distributionFile));
+
+        XPath xPath = XPathFactory.newInstance().newXPath();
+
+        String v = (String) xPath.evaluate(
+                    "/installer-gui-script/options/@hostArchitectures",
+                    doc, XPathConstants.STRING);
+
+        if ("aarch64".equals(System.getProperty("os.arch"))) {
+            TKit.assertEquals(v, "arm64",
+                    "Check value of \"hostArchitectures\" attribute");
+        } else {
+            TKit.assertEquals(v, "x86_64",
+                    "Check value of \"hostArchitectures\" attribute");
+        }
+    }
+
+    @Test
+    public static void test() {
+        new PackageTest()
+                .excludeTypes(PackageType.MAC_DMG) // Not valid case
+                .configureHelloApp()
+                .forTypes(PackageType.MAC_PKG)
+                .addInstallVerifier(HostArchPkgTest::verifyHostArch)
+                .run();
```

Comment:
The test is applicable only to the scenario when .pkg installer is unpacked and not when it is installed. So `PackageTest.run()` is not quite a good fit for this execution scenario as it depends on the value of `jpackage.test.action` system property (its default value is indeed to create and unpack installer, but can be overriden). The better option would be to use `PackageTest.run()` with explicit list of actions the test should perform. Suggested fix:
```
new PackageTest()
                .forTypes(PackageType.MAC_PKG)
                .configureHelloApp()
                .addInstallVerifier(HostArchPkgTest::verifyHostArch)
                .run(PackageTest.Action.CREATE_AND_UNPACK);
```

Reply:
Sorry for the inconvenience, I didn't include this comment in my initial review.